crates/cubecl-runtime/src/client.rs

```diff
@@ -349,17 +383,71 @@ where
     }
 
     /// Measure the execution time of some inner operations.
-    ///
-    /// Nb: this function will only allow one function at a time to be submitted when multi threading.
-    /// Recursive measurements are not allowed and will deadlock.
-    pub fn profile(&self, func: impl FnOnce()) -> ProfileDuration {
+    #[track_caller]
+    pub fn profile(
+        &self,
+        func: impl FnOnce(),
+        #[allow(unused)] func_name: &str,
+    ) -> ProfileDuration {
+        // Get the outer caller. Maybe this should be lazy, idk.
+        // For execute() this could also maybe point straight to the
+        // cube kernel instead. For general profiling that's not possible.
+        #[cfg(feature = "profile-tracy")]
```

Comment:
You can profile multiple kernels at the same time, so pointing to a kernel location isn't really possible or desirable.